src/libYARP_os/src/yarp/os/impl/NameConfig.cpp

```diff
@@ -90,7 +91,7 @@ std::string NameConfig::expandFilename(const char* fname)
     std::string root = ResourceFinder::getConfigHome();
     std::string conf;
     if (!root.empty()) {
-        conf = root + NetworkBase::getDirectorySeparator() + fname;
+        conf = root + std::string{yarp::conf::filesystem::preferred_separator} + fname;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'root' is reassigned a value before the old one has been used.](https://app.codacy.com/app/robotology/yarp/pullRequest?prid=3865571)